src/literals.py

```diff
@@ -33,3 +36,42 @@ class Ports:
     "SASL_SSL": Ports(9093, 19093),
     "SSL": Ports(9094, 19094),
 }
+
+AvailableStatuses = Literal[
+    "ACTIVE",
+    "SNAP_NOT_INSTALLED",
+    "ZK_NOT_RELATED",
+    "ZK_NOT_CONNECTED",
+    "SNAP_NOT_RUNNING",
+    "ADDED_STORAGE",
+    "REMOVED_STORAGE",
+    "REMOVED_STORAGE_NO_REPL",
+    "ZK_TLS_MISMATCH",
+    "NO_PEER_RELATION",
+    "ZK_NO_DATA",
+    "NO_BROKER_CREDS",
+    "NO_CERT",
+]
+
+
+STATUS: Dict[AvailableStatuses, StatusBase] = {
+    "ACTIVE": ActiveStatus(),
+    "NO_PEER_RELATION": WaitingStatus("no peer relation yet"),
+    "SNAP_NOT_INSTALLED": BlockedStatus(f"unable to install {SNAP_NAME} snap"),
+    "SNAP_NOT_RUNNING": BlockedStatus("snap service not running"),
+    "ZK_NOT_RELATED": BlockedStatus("missing required zookeeper relation"),
+    "ZK_NOT_CONNECTED": BlockedStatus("unit not connected to zookeeper"),
+    "ZK_TLS_MISMATCH": BlockedStatus("tls must be enabled on both kafka and zookeeper"),
+    "ZK_NO_DATA": WaitingStatus("zookeeper credentials not created yet"),
+    "ADDED_STORAGE": ActiveStatus(
+        "manual partition reassignment may be needed to utilize new storage volumes"
+    ),
+    "REMOVED_STORAGE": BlockedStatus(
+        "manual partition reassignment from replicated brokers recommended due to lost partitions on removed storage volumes"
+    ),
```

Comment:
It will remove on the next `update-status`, which isn't ideal as it might get lost. 

I think best-case scenario, is that we'd manually remove it when the charm has some sort of built-in action for re-assigning partitions, but in the absence of this is mostly harmless. 

What do you think?

Reply:
I'm actually not sure. I actually would have simply put this message in the logs, as a warning level. I understand that using statuses is probably more evident to the user, but I'm a bit concerned that this may create confusion: I'm wondering whether "Blocked" is correct since BlockedStatus should stay there unless the user does something to resolve this. I understand this is a bit borderline, since the user SHOULD do something, but we don't have a way to assess whether he has done it.

In opensearch they rely on a "health" endpoint, provided by open-search that they constatntly perform check against. I'm wondering whether we could do something similar: like implementing a check (even using high-level property as you have done here) that each topics has correct partitions, and if not notify this to the user via statuses messages. In any update status then, we would keep having BlockedStatus until the user has correctly perform partition re-assignment. 

This would be of course a new ticket, not to be done within this PR

Reply:
Maybe @grobbie can also provide some inputs, since it is somewhat UX related (or @delgod or @Mehdi-Bendriss since they possibly faced similar thoughts in opensearch)

Reply:
The 'health' endpoint is actually what I was trying to mimic with the `on_update_status` handler that checks `if self.healthy():`.

i.e with default model settings, every 5 minutes we ping Kafka, if it's up, set `ActiveStatus`. I may be mistaken, but `BlockedStatus` can actually get overwritten by the charm, which might happen here.

When we scale units/storage, the Kafka service will handle itself just fine, so it will still be 'healthy', just not from a product perspective. I can also add a warning log here too as it's fairly critical.

Reply:
logs are less transitive, and can be scraped and properly alerted with automation